src/sentry/dynamic_sampling/prioritise_projects.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence, Tuple
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry import options
+from sentry.dynamic_sampling.rules.utils import OrganizationId, ProjectId
+from sentry.sentry_metrics import indexer
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 1000
+
+
+def fetch_projects_with_total_volumes() -> Mapping[OrganizationId, Sequence[Tuple[ProjectId, int]]]:
+    """
+    This function fetch with pagination orgs and projects with count per root project
+    """
+    aggregated_projects = defaultdict(list)
+    start_time = time.time()
+    offset = 0
+    sample_rate = int(options.get("dynamic-sampling.prioritise_projects.sample_rate") * 100)
+    metric_id = indexer.resolve_shared_org(str(TransactionMRI.COUNT_PER_ROOT_PROJECT.value))
+    while (time.time() - start_time) < MAX_SECONDS:
+        query = (
+            Query(
+                match=Entity(EntityKey.GenericOrgMetricsCounters.value),
+                select=[
+                    Function("sum", [Column("value")], "root_count_value"),
+                    Column("org_id"),
+                    Column("project_id"),
+                ],
+                groupby=[Column("org_id"), Column("project_id")],
+                where=[
+                    Condition(Function("modulo", [Column("org_id"), 100]), Op.LT, sample_rate),
```

Comment:
This might work fine but I'm not sure how efficient `WHERE org_id % 100 < 45` will be at scanning the table.

@nikhars do you have an opinion on this? I think given the data size and the filtering by metric_id it would probably work but I wonder if they'd get better ClickHouse performance if they enumerated the org_id's to check into batches of 5k or 10k and filtering before sending the query